.github/workflows/test.yml

```diff
@@ -11,10 +11,31 @@ jobs:
       max-parallel: 5
       matrix:
         python-version: ['3.8', '3.9', '3.10', '3.11', '3.12-dev']
-        django-version: ['3.2', '4.0', 'main']
+        django-version: ['3.2', '4.0', '4.1', '4.2', 'main']
         include:
           - python-version: '3.7'
             django-version: '3.2'
+        exclude:
+          - python-version: '3.11'
+            django-version: '3.2'
+          - python-version: '3.12-dev'
+            django-version: '3.2'
+
+          - python-version: '3.11'
+            django-version: '4.0'
+          - python-version: '3.12-dev'
+            django-version: '4.0'
+
+          - python-version: '3.12-dev'
+            django-version: '4.1'
+
+          - python-version: '3.8'
+            django-version: 'main'
+          - python-version: '3.9'
+            django-version: 'main'
+          - python-version: '3.10'
+            django-version: 'main'
+
```

Comment:
We ought to drop unsupported versions of Python. We can do that separately though, to avoid growing the scope on this PR.

See: https://github.com/jazzband/django-user-sessions/issues/169

Reply:
I'm fine with this, but to lessen the risk of somebody objecting to this PR I didn't want to remove anything that others might be relying on. I find that removing support for old versions is best done in a PR that does _only_ that.